lexer/impl/lexer.cpp

```diff
@@ -1,6 +1,309 @@
+        trie.Add("end", TokenType::End);
+        trie.Add("reverse", TokenType::ReverseRange);
+        trie.Add("in", TokenType::InRange);
+        trie.Add("if", TokenType::If);
+        trie.Add("then", TokenType::Then);
+        trie.Add("else", TokenType::Else);
+        trie.Add("and", TokenType::AndLogic);
+        trie.Add("or", TokenType::OrLogic);
+        trie.Add("xor", TokenType::XorLogic);
+        // comparison
+        trie.Add("<", TokenType::SmallerComp);
+        trie.Add("<=", TokenType::SeqComp);
+        trie.Add(">", TokenType::BiggerComp);
+        trie.Add(">=", TokenType::BeqComp);
+        trie.Add("=", TokenType::EqComp);
+        trie.Add("/=", TokenType::NeqComp);
+        // math ops
+        trie.Add("*", TokenType::MultOp);
+        trie.Add("/", TokenType::DivOp);
+        trie.Add("%", TokenType::RemainderOp);
+        trie.Add("+", TokenType::PlusOp);
+        trie.Add("-", TokenType::MinusOp);
+        // signs
+        trie.Add(".", TokenType::Dot);
+        trie.Add("..", TokenType::TwoDots);
+        trie.Add(",", TokenType::Comma);
+        // braces and parenthesis
+        trie.Add("(", TokenType::BracketOpen);
+        trie.Add(")", TokenType::BracketClose);
+        trie.Add("[", TokenType::SquareBracketOpen);
+        trie.Add("]", TokenType::SquareBracketClose);
+        trie.Add("//", TokenType::SingleLineComment);
+        trie.Add("\n", TokenType::NewLine);
+        trie.Add(";", TokenType::Semicolumn);
+        trie.Add(":=", TokenType::Assignment);
+        trie.Add(";", TokenType::Column);
+    }
+
+    /**
+     * Checks of the longToken is in the word from position *indexPoiter
+     * UPDATES the index, in case of full equality
+     * ADDS required tokens to the result vector
+     * 
+     * @param indexPointer - pointer to int index in the word
+     * @param word - the word, we are searching at
+     * @param longToken - string of length 2, e.g. ">=", "<="...
+     * @param tt1 - TokenType if only the first chars are equal
+     * @param tt2 - TokenType if all chars are equal
+     * 
+    */
+    void checkTwoCharToken(int *indexPointer, std::string word, std::string longToken,
+                           TokenType tt1, TokenType tt2)
+    {
+        char firstChar = longToken[0];
+        if (*indexPointer < int(word.length()) && word[*indexPointer] == firstChar)
+        {
+            (*indexPointer)++;
+            result.push_back(makeToken(tt2, lineNum, longToken));
+        }
+        else
+        {
+            result.push_back(makeToken(tt1, lineNum, toStr(firstChar)));
+        }
+    }
+
+    /**
+     * Itterates throught word while it is a number ()
+     * 
+    */
+    std::string readWhileNum(int *indexPointer, std::string word)
+    {
+        std::string num;
+        while (*indexPointer < int(word.length()) && isdigit(word[*indexPointer]))
+        {
+            char c = word[*indexPointer];
+            num.append(toStr(c));
```

Comment:
Unneeded string initializations and copies. Just use `push_back`
```suggestion
num.push_back(c);
```